client/src/light/backend.rs

```diff
@@ -284,18 +285,17 @@ where
 		check_genesis_storage(&top, &children)?;
 
 		// this is only called when genesis block is imported => shouldn't be performance bottleneck
-		let mut storage: HashMap<Option<Vec<u8>>, StorageOverlay> = HashMap::new();
+		let mut storage: HashMap<Option<(Vec<u8>, OwnedChildInfo)>, StorageOverlay> = HashMap::new();
```

Comment:
Usage of 'OwnedChildInfo' in the key is rather questionable here, yet it seem to work well with current implementation.